standards/bulk.md

```diff
@@ -0,0 +1,134 @@
+            "ref": "sps-ref" | null,                                        // the associated sps-ref URN entity, if applicable
+            "status": "SUCCEEDED" | "FAILED",                               // status of individual operation
+            "detail": "string",                                             // indicates details on the operation result, such as a error message.
+        }
+    ]
+}
+```
+
+The following example demonstrates various operations in a bulk request and response:
+
+```
+// REQUEST
+PATCH /articles
+Content-Type: application/json
+{
+    "operations": [
+        {
+            "action": "CREATE_UPDATE"
+            "ifMatch": "33a64df551425fcc55e4d42a148795d9f25f89d4",
+            "entity": {
+                "id": "bfd8f0c0-be67-4f81-bf82-e55e552609f4",
+                "name": "my name",
+                "description": "my description"
+            }
+        },
+        {
+            "action": "CREATE"  
+            "entity": {
+                "id": null,
+                "name": "my name",
+                "description": "my description"
+            }
+        },
+        {
+            "action": "DELETE" 
+            "ifMatch": "44a64df551425fcc55e4d42a148795d9f25f89c5",     
+            "entity": {
+                "id": "d9bd5d91-fc25-4410-ae42-c8f631e8e9ff",
+                "name": null,
+                "description": null
+            }
+        },
+    ]
+}
+ 
+// RESPONSE
+200 OK
+Content-Type: application/json
+{
+    "status": "PARTIAL",                                 
+    "operations": [
+        {
+            "ordinal": 0,
+            "action": "CREATE_UPDATE",
+            "id": "bfd8f0c0-be67-4f81-bf82-e55e552609f4",
+            "ref": "sps:...",
+            "status": "SUCCEEDED",                         
+            "detail": "Article was updated.",  
+        },
+        {
+            "ordinal": 1,
+            "action": "CREATE",
+            "id": null,
+            "ref": null,
+            "status": "FAILED",                         
+            "detail": "Could not create article, since the name already exists.",  
+        },
+        {
+            "ordinal": 2,
+            "action": "DELETE",
+            "id": "d9bd5d91-fc25-4410-ae42-c8f631e8e9ff",
+            "ref": "sps:...",
+            "status": "SUCCEEDED",                         
+            "detail": null,  
+        }
+    ]
+}
+```
+
+```note
+**PATCH REQUEST ATOMICITY**
+While the [HTTP PATCH RFC](https://datatracker.ietf.org/doc/html/rfc5789) requires full atomicity in application of the patched document, this is not possible in the case of bulk operations. In the case of bulk operations, the request body is a collection of resources which means that the request body is not a single document. While not a perfect interpretation of PATCH execution, the boundaries of bulk operations require some compromise to benefit developer experience and consistency.
```

Comment:
how true is this? how confident are we? https://datatracker.ietf.org/doc/html/rfc5789
```
A PATCH request can be issued in such a way as to be idempotent,
   which also helps prevent bad outcomes from collisions between two
   PATCH requests on the same resource in a similar time frame.
   Collisions from multiple PATCH requests may be more dangerous than
   PUT collisions because some patch formats need to operate from a
   known base-point or else they will corrupt the resource.  Clients
   using this kind of patch application SHOULD use a conditional request
   such that the request will fail if the resource has been updated
   since the client last accessed the resource.  For example, the client
   can use a strong ETag [[RFC2616](https://datatracker.ietf.org/doc/html/rfc2616)] in an If-Match header on the PATCH
   request.

   There are also cases where patch formats do not need to operate from
   a known base-point (e.g., appending text lines to log files, or non-
   colliding rows to database tables), in which case the same care in
   client requests is not needed.

   The server MUST apply the entire set of changes atomically and never
   provide (e.g., in response to a GET during this operation) a
   partially modified representation.  If the entire patch document
   cannot be successfully applied, then the server MUST NOT apply any of
   the changes.  The determination of what constitutes a successful
   PATCH can vary depending on the patch document and the type of
   resource(s) being modified.  For example, the common 'diff' utility
   can generate a patch document that applies to multiple files in a
   directory hierarchy.  The atomicity requirement holds for all
   directly affected files.  See "Error Handling", [Section 2.2](https://datatracker.ietf.org/doc/html/rfc5789#section-2.2), for
   details on status codes and possible error conditions.
```

While odd, JSON Patch also abides by this atomicity and respects the HTTP PATCH (https://datatracker.ietf.org/doc/html/rfc6902/). But it really is how you interpret the payload here. Idempotency is not a requirement for PATCH (would be nice).

Reply:
We will be supporting both atomic and isolated transaction behavior with isolated being the default, regardless of the exact PATCH specification due to further implications.